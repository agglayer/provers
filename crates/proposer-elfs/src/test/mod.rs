use alloy_primitives::B256;
use sp1_sdk::HashableKey;

use super::*;

#[rstest::rstest]
#[case::agg(aggregation::ELF, &aggregation::VKEY, aggregation::VKEY_HASH)]
#[case::range(range::ELF, &range::VKEY, range::VKEY_HASH)]
fn consistency(#[case] elf: &[u8], #[case] vkey: &LazyVerifyingKey, #[case] vkey_hash: VKeyHash) {
    let prover = sp1_sdk::CpuProver::new();
    let (_proving_key, computed_vkey) = sp1_sdk::Prover::setup(&prover, elf);

    assert_eq!(VKeyHash::from_vkey(&computed_vkey), vkey_hash);
    assert_eq!(VKeyHash::from_vkey(vkey.vkey()), vkey_hash);
}

#[test]
fn range_commitment_consistency() {
    assert_eq!(range::VKEY.hash_bytes(), range::VKEY_COMMITMENT);
}

#[rstest::rstest]
#[case::agg("aggregation", &aggregation::VKEY)]
#[case::range("range", &range::VKEY)]
fn snap_vkey_hash(#[case] name: &'static str, #[case] vkey: &LazyVerifyingKey) {
    let hu32 = vkey.hash_u32();
    let bytes32 = B256::new(vkey.bytes32_raw());
    let bytes = B256::new(vkey.hash_bytes());

    let snap = format!("{name} vkey\nhash_u32 {hu32:?}\nbytes32  {bytes32}\nbytes    {bytes}\n");

    insta::assert_snapshot!(format!("{name}_vkey"), snap);
}
