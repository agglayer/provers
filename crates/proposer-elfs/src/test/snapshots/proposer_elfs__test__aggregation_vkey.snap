---
source: crates/proposer-elfs/src/test/mod.rs
expression: snap
snapshot_kind: text
---
aggregation vkey
hash_u32 [1638338388, 1256426946, 526664290, 1122453194, 1403846808, 601906877, 1176579776, 312933952]
bytes32  0x00c34e1ea92b8e3708fb2213142e746caa75a01308f817af6310976012a6fe40
bytes    0x61a70f544ae38dc21f64426242e746ca53ad009823e05ebd46212ec012a6fe40
